File: services/auth-service/src/domain/repositories/ISedeRepository.ts
import { Sede } from '../entities/Sede';

/**
 * Puerto (interfaz) del repositorio de sedes.
 */
export interface ISedeRepository {
  findAll(): Promise<Sede[]>;
  findById(id: number): Promise<Sede | null>;
  findByCodigo(codigo: string): Promise<Sede | null>;
  findByRegion(regionId: number): Promise<Sede[]>;
}

File: services/auth-service-py/src/domain/repositories/i_sede_repository.py
from abc import ABC, abstractmethod
from typing import Optional
from ..entities.sede import Sede


class ISedeRepository(ABC):

    @abstractmethod
    async def find_all_active(self) -> list[Sede]:
        ...

    @abstractmethod
    async def find_by_id(self, sede_id: int) -> Optional[Sede]:
        ...

    @abstractmethod
    async def find_by_codigo(self, codigo: str) -> Optional[Sede]:
        ...
